### ISSUE TITLE
Loading pages leaves the search bar on the screen.

### ISSUE THREAD

**Issue by jrmuizel** [mozilla-mobile/firefox-ios#18697]
## Steps to reproduce

- Tap on the url bar
- Tap on one of the favicons (like the one for the Toronto Star)

See 

https://github.com/mozilla-mobile/firefox-ios/assets/332653/886e812f-e84f-40d4-96df-785f8e7f69f3

### Expected behavior

The new page should load, the keyboard should go away and the url bar should have a url for the current page.

### Actual behavior

The new page loads, the keyboard stays up and url bar has the url for the previous page.

### Device & build information
* Device: 14 Mini
* Build version: ? 122.2
* First seen version: ? Recently

### Notes

This seems to only happen when in the control group of "We should prefer switching to open tabs vs..." experiment.

Attachments:

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Diana Andreea Barladeanu Andrei Bodea can you reproduce it? How often is happening? Thanks!

**Comment by afurlan-firefox** [comment]
@jrmuizel can you confirm you iOS version? I am not able to reproduce. Thanks!

**Comment by jrmuizel** [comment]
I'm on an iPhone 13 mini (not 14 mini like I wrote) on iOS 17.2.1

**Comment by jrmuizel** [comment]
> @jrmuizel can you confirm you iOS version? I am not able to reproduce. Thanks!

Did you confirm that you were in the control group of the experiment?

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Yes, I am. I was able to repro, but it only happens to me if I stay on the same page (I am in the tab “google.com ( http://google.com )” and I click on the top site “google.com ( http://google.com )”). If I switch to other top site, it hides the keyboard and unfocus the URL bar. Is that your case too? I just want to make sure I got the whole error. Thanks!

[^RPReplay_Final1707855106.MP4]

[^RPReplay_Final1707855176.MP4]

**Comment by jrmuizel** [comment]
Perhaps you need to have an existing tab that's being replaced for it to reproduce?

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

You are right! Thanks.

**Comment by jrmuizel** [comment]
Yes this seems to be the case. To reproduce make sure you have a tab that's already open for the same page.

e.g. if I have two tabs
- news.ycombinator.com
- thestar.com

I switch to the news.ycombinator.com tab and then click on the icon for the thestar.com

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Fixed on FX 123

**Comment by jrmuizel** [comment]
What change fixed it?
